types/config.go

```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"encoding/json"
+)
+
+// VMConfig defines the configuration for the VM.
+// For full documentation see the Rust side:
+// https://docs.rs/cosmwasm-vm/2.2.0-rc.1/cosmwasm_vm/struct.Config.html
+type VMConfig struct {
+	WasmLimits WasmLimits   `json:"wasm_limits"`
+	Cache      CacheOptions `json:"cache"`
+}
+
+// WasmLimits defines limits for static validation of wasm code that is stored on chain.
+// For full documentation see the Rust side:
+// https://docs.rs/cosmwasm-vm/2.2.0-rc.1/cosmwasm_vm/struct.WasmLimits.html
+type WasmLimits struct {
+	InitialMemoryLimit     *uint32 `json:"initial_memory_limit,omitempty"`
+	TableSizeLimit         *uint32 `json:"table_size_limit,omitempty"`
```

Comment:
Also similar question here. Is 4GiB enough?

Reply:
Yes. 4GB as the memory limit is definitely enough since Wasm is (discounting the memory64 proposal) a 32-bit virtual machine, meaning it can only ever address 4GB of memory (that's why we pretty quickly skedaddled away to 64-bits on our physical machines.

Reply:
The memory64 proposal will change that and make memory addressable with 64bit indexes, but I'm not sure how close we even are in CosmWasm to enable that proposal.

Reply:
See <https://webassembly.org/features/>

Reply:
Yup, as aumetra mentioned, for now wasm only supports 4GB and I don't see a very legitimate reason to allow a contract to use anywhere close to that much memory (for reference: `InstanceMemoryLimit` is currently hardcoded to 32 MB by wasmd).
However, `InitialMemoryLimit` is measured in wasm pages of 64KiB, so this limit can already be set quite high. I guess this underlines your point about making the unit more clear. I only added it to the docs, which are linked at the top of the struct: https://docs.rs/cosmwasm-vm/2.2.0-rc.1/cosmwasm_vm/struct.WasmLimits.html#structfield.initial_memory_limit

Reply:
Regarding table size: That's just the number of elements, not bytes. Wasm tables contain references (to functions or host values). Maybe `_elements` as a suffix?